docs/Contributors_Guide/code_profiling.rst

```diff
@@ -0,0 +1,538 @@
+  (i.e. no stress-testing support is available).
+
+
+
+Overview
+========
+
+Customizations for MET
+----------------------
+
+A Python script, benchmark.py is available to exercise the MET source code under consideration
+either via MET commands (to replicate command line usage of the MET tool), or via METplus use
+case (utilizing the METplus wrapper code and associated configuration files). The Python script consolidates the
+summary and detail metrics information (as text files) into csv and tabular text files. The benchmark.py
+script has an accompanying configuration file, benchmark.yaml located in the
+*$HOME/MET/internal/scripts/benchmark* directory (where \$HOME is the directory where the MET source code is located).
+
+The **ctrack.hpp** header file is modified to allow the summary and detailed reports to be saved as text files to
+facilitate the consolidation of information into csv and tabular formats. The summary and detail metrics files are
+located in the directory where the benchmark.py script was invoked.  The modified version of ctrack.hpp is located
+in the *\${BASE_DIR}/MET/src/basic/vx_util* directory, where ${BASE_DIR} is the full path to where the MET source code
+has been cloned or forked.
+
+Code that is currently instrumented
+-----------------------------------
+
+The following code is instrumented using CTRACK:
+
+- MET/src/basic/vx_util/main.cpp
+   - do_pre_process function
+   - do_post_process function
+- MET/src/tools/core/ensemble_stat/ensemble_stat.cc
+- MET/src/tools/core/ensemble_stat/ensemble_stat_conf.cc
+
+Benchmarking with Python script
+-------------------------------
+
+The benchmarking.py script invokes MET code either via **MET command line commands** or **METplus use cases** as
+specified by the **run_met_directly** setting in the benchmark.yaml configuration file.
+The metrics from the summary and detail tables are consolidated into csv and tabular text files
+(the locations of these consolidated metrics text files are specified in the benchmark.yaml configuration file).
+The CTRACK summary_output.txt and detail_output.txt reports (containing the performance metrics) are written to the directory
+from which the benchmarking.py script was executed.  An information file is also generated that captures the version
+of Python used, a timestamp, and any other relevant information for capturing the environment under which the code was
+profiled/benchmarked.
+
+Overview of Steps for Performing Benchmarking
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. .. dropdown:: Instrument the MET code of interest
+
+       .. note::
+
+         The ctrack.hpp file is saved in the $HOME/MET/src/basic/vx_util directory and does not need to be modified
+         or added to any other location.  This version of ctrack.hpp has been modified to write the summary and detail
+         tables to text files. By default, CTRACK is disabled and is enabled at compilation time via the
+         **--enable-profiler flag**.
```

Comment:
```suggestion
         :code:`--enable-profiler` flag.
```
The `--enable-profiler` doesn't render well in the docs. Recommend using code formatting.